functions

```diff
@@ -857,20 +857,22 @@ install_modules() {
 
     cp "$@" "$moduledest/kernel"
 
-    # unzip modules prior to recompression
-    for m in "$@"; do
-        case $m in
-            *.xz)
-                xz_comp+=("$moduledest/kernel/${m##*/}")
-                ;;
-            *.gz)
-                gz_comp+=("$moduledest/kernel/${m##*/}")
-                ;;
-            *.zst)
-                zst_comp+=("$moduledest/kernel/${m##*/}")
-                ;;
-        esac
-    done
+    if [[ $MODULES_DECOMPRESS == yes ]]; then
```

Comment:
```suggestion
    if [[ "$MODULES_DECOMPRESS" = yes ]]; then
```

Reply:
No! `=` is the POSIX syntax. Bashisms, such as `==`, should be prefered.

Reply:
1. Is there some sort of style guide which recommends using always using bashisms? I've never heard any style guide dictate any such rationale.
2. We can probably all agree you shouldn't use `[[ ! a == b ]]` over [[ a != b ]]` just to use more bashisms, right? bashisms are good to use, when there is value in their use; they shouldn't be used just to use them. That's cargo culting.
3. `[[ a = b ]]` _is_ a bashism. `=` without context is just a char. `[[` is a bashism, so all `[[` operators are bashisms.
4. `help [[` says `When the `==' and `!=' operators are used, the string to the right of the operator is used as a pattern and pattern matching is performed.` Compare to `help test` and `STRING1 = STRING2  True if the strings are equal.` which is different.

Reply:
The project guidelines says to generally prefer bash syntax. And I agree.

https://github.com/archlinux/mkinitcpio/blob/7bfe4861eacb3bf6cb70d9a17a0262542733a8ed/HACKING#L7-L11

Since `[[` is different from `[`/`test`, `help test` doesn't apply to it. `[[` will always perform pattern matching.